### ISSUE TITLE
Crashing on trying to change settings.

### ISSUE THREAD

**Issue by Lord-Kamina** [ozguncagri/nwMAME#2]
I've currently pointed the app to my roms directory (which is relatively big; contains 10492 roms), and also added a cover folder that should cover all, if not most, of these. The app first crashed when I tried to update the rom list; eventually I managed to do that, but now I cannot change the cover type from "List" to anything else as, after a while, the app crashes.

The obvious guess would be it's not being able to cope with the large database or something like that.

Here I'm attaching a crash report...

https://www.dropbox.com/s/c6e2a6vflvgylnj/nwjs%20Helper_2015-08-16-200326_Tiamat.crash?dl=0

**Comment by ozguncagri** [comment]
Hmm. You are the first person who tries with such a big library. I don't even have that much game at all :) The crash is probably the same reason. Application tries to load all games into the app's DOM. And that is probably such a big data for this kind of web app. I'll test it with all MAME games and I'll try to optimize and fix it for large databases as soon as possible.

Thanks for report btw...

**Comment by Lord-Kamina** [comment]
Yeah, I figure... and also the fact that the way the database is currently organized... the whole DB basically ends up inside a BLOB field, that's hardly optimal.

I also noticed that not all covers will load in in "large" mode (perhaps in small also, but haven't tried that mode) although I am chalking it up to the same thing. Probably the DOM is so cluttered it has trouble loading all images.

BTW, so far what I'm using for covers is the MAME Titles pack (you can find them for download if you google, basically screens of the title screen for most if not every games.)
